cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -929,7 +930,27 @@ ReconciliationState createReconciliationState(Reconciliation reconciliation, Kaf
         }
 
         Future<ReconciliationState> zkNetPolicy() {
-            return withVoid(networkPolicyOperator.reconcile(namespace, ZookeeperCluster.policyName(name), zkCluster.generateNetworkPolicy()));
+            Future future = Future.future();
+            networkPolicyOperator.reconcile(namespace, ZookeeperCluster.policyName(name), zkCluster.generateNetworkPolicy(true)).setHandler(res -> {
```

Comment:
`Future.recover()` would be a more elegant way to do this.

Reply:
TBH, I'm not sure I understand how would that work. It doesn't seem to handle the successful case which we need to wrap into the new future as a succeeded.

Reply:
Ok, should be fixed. Thanks